include/hybridmap/HybridNeutralAtomMapper.hpp

```diff
@@ -0,0 +1,534 @@
+  }
+
+  /**
+   * @brief Prints the mapped circuits as an extended OpenQASM string.
+   * @return The mapped quantum circuit with abstract SWAP gates and MOVE
+   */
+  [[maybe_unused]] std::string getMappedQc() {
+    std::stringstream ss;
+    this->mappedQc.dumpOpenQASM(ss, false);
+    return ss.str();
+  }
+
+  /**
+   * @brief Saves the mapped quantum circuit to a file.
+   * @param filename The name of the file to save the mapped quantum circuit to
+   */
+  [[maybe_unused]] void saveMappedQc(const std::string& filename) {
+    std::ofstream ofs(filename);
+    this->mappedQc.dumpOpenQASM(ofs, false);
+  }
+
+  /**
+   * @brief Prints the mapped circuit with AOD operations as an extended
+   * OpenQASM
+   * @return The mapped quantum circuit with native AOD operations
+   */
+  [[maybe_unused]] std::string getMappedQcAOD() {
+    std::stringstream ss;
+    this->mappedQcAOD.dumpOpenQASM(ss, false);
+    return ss.str();
+  }
+
+  /**
+   * @brief Saves the mapped quantum circuit with AOD operations to a file.
+   * @param filename The name of the file to save the mapped quantum circuit
+   * with AOD operations to
+   */
+  [[maybe_unused]] void saveMappedQcAOD(const std::string& filename) {
+    std::ofstream ofs(filename);
+    this->mappedQcAOD.dumpOpenQASM(ofs, false);
+  }
+
+  /**
+   * @brief Schedules the mapped quantum circuit on the neutral atom
+   * architecture.
+   * @details For each gate/operation in the input circuit, the scheduler checks
+   * the earliest possible time slot for execution. If the gate is a multi qubit
+   * gate, also the blocking of other qubits is taken into consideration. The
+   * execution times are read from the neutral atom architecture.
+   * @param verboseArg If true, prints additional information
+   * @param createAnimationCsv If true, creates a csv file for the animation
+   * @param shuttlingSpeedFactor The factor to speed up the shuttling time
+   * @return The results of the scheduler
+   */
+  [[maybe_unused]] SchedulerResults schedule(bool verboseArg           = false,
+                                             bool createAnimationCsv   = false,
+                                             fp   shuttlingSpeedFactor = 1.0) {
+    scheduler = NeutralAtomScheduler(arch);
+    return scheduler.schedule(mappedQcAOD, hardwareQubits.getInitialHwPos(),
+                              verboseArg, createAnimationCsv,
+                              shuttlingSpeedFactor);
+  }
+
+  /**
+   * @brief Saves the animation csv file of the scheduled quantum circuit.
+   * @return The animation csv string
+   */
+  [[maybe_unused]] std::string getAnimationCsv() {
+    return scheduler.getAnimationCsv();
+  }
+
+  /**
+   * @brief Saves the animation csv file of the scheduled quantum circuit.
+   * @param filename The name of the file to save the animation csv file to
+   */
+  [[maybe_unused]] void saveAnimationCsv(const std::string& filename) {
+    scheduler.saveAnimationCsv(filename);
+  }
```

Comment:
GIven the comment above, these could be removed.

Reply:
See below